datastore/providers/pgvector_datastore.py

```diff
@@ -0,0 +1,122 @@
+import os
+from typing import Dict, List, Optional
+
+from pgvector.sqlalchemy import Vector
+from sqlalchemy import Column, Index, String, create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.future import select
+from sqlalchemy.orm import sessionmaker
+
+from datastore.datastore import DataStore
+from models.models import (
+    Document,
+    DocumentChunk,
+    DocumentMetadataFilter,
+    QueryResult,
+    QueryWithEmbedding,
+)
+
+Base = declarative_base()
+
+# Read environment variables for pgvector configuration
+PGVECTOR_COLLECTION = os.getenv("PGVECTOR_COLLECTION", "documents")
+PGVECTOR_URL = os.getenv("PGVECTOR_URL")
+
+
+class VectorDocument(DataStore):
+    __tablename__ = PGVECTOR_COLLECTION
+
+    id = Column(String, primary_key=True)
+    document_id = Column(String)
+    text = Column(String)
+    embedding = Column(Vector(1536))
```

Comment:
Can you please add support for the other optional metadata fields in the `DocumentMetadata` model so that this is consistent with the other vector db implementations? Otherwise this can't be merged.